misk/src/test/kotlin/misk/web/ContentBasedDispatchTest.kt

```diff
@@ -0,0 +1,133 @@
+package misk.web
+
+import com.google.inject.util.Modules
+import com.squareup.moshi.Moshi
+import misk.MiskModule
+import misk.inject.KAbstractModule
+import misk.testing.ActionTest
+import misk.testing.ActionTestModule
+import misk.testing.TestWebModule
+import misk.web.actions.WebAction
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import misk.web.mediatype.asMediaType
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@ActionTest(startService = true)
+internal class ContentBasedDispatchTest {
+    @ActionTestModule
+    val module = Modules.combine(
+            MiskModule(),
+            WebModule(),
+            TestWebModule(),
+            TestModule())
+
+    data class Packet(val message: String)
+
+    private val jsonMediaType = MediaTypes.APPLICATION_JSON.asMediaType
+    private val plainTextMediaType = MediaTypes.TEXT_PLAIN_UTF8.asMediaType
+    private val weirdTextMediaType = "text/weird".asMediaType
+
+    private @Inject lateinit var moshi: Moshi
+    private @Inject lateinit var jettyService: JettyService
+    private val packetJsonAdapter get() = moshi.adapter(Packet::class.java)
+
+    @Test
+    fun postJsonExpectJson() {
+        val requestContent = packetJsonAdapter.toJson(Packet("my friend"))
+        val responseContent = post(jsonMediaType, requestContent, jsonMediaType).source()
+        assertThat(packetJsonAdapter.fromJson(responseContent)!!.message)
+                .isEqualTo("json/json my friend")
```

Comment:
first read through I read the `/` as the media type separator, like `application/json`. A comma would be a little less surprising, or even `"request=json response=json my friend"`